deploy/charts/firefly/templates/_helpers.tpl

```diff
@@ -0,0 +1,153 @@
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "firefly.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "firefly.fullname" -}}
+{{- if .Values.fullnameOverride }}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- $name := default .Chart.Name .Values.nameOverride }}
+{{- if contains $name .Release.Name }}
+{{- .Release.Name | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{/*
+Create chart name and version as used by the chart label.
+*/}}
+{{- define "firefly.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+Common labels
+*/}}
+{{- define "firefly.labels" -}}
+helm.sh/chart: {{ include "firefly.chart" . }}
+{{ include "firefly.selectorLabels" . }}
+{{- if .Chart.AppVersion }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+{{- end }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- end }}
+
+{{/*
+Selector labels
+*/}}
+{{- define "firefly.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "firefly.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end }}
+
+{{/*
+Create the name of the service account to use
+*/}}
+{{- define "firefly.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create }}
+{{- default (include "firefly.fullname" .) .Values.serviceAccount.name }}
+{{- else }}
+{{- default "default" .Values.serviceAccount.name }}
+{{- end }}
+{{- end }}
+
+{{- define "firefly.dataexchangeP2PHost" -}}
+{{- if .Values.dataexchange.ingress.enabled }}
+{{- index .Values.dataexchange.ingress.hosts 0 }}
+{{- else }}
+{{- printf "%s-dx:%d" (include "firefly.fullname" .) (.Values.dataexchange.service.p2pPort | int64) }}
+{{- end }}
+{{- end }}
+
+{{- define "firefly.core.config" -}}
+{{- if .Values.config.debugEnabled }}
+log:
+  level: debug
+debug:
+  port: 6060
+{{- end }}
+http:
+  port: 5000
```

Comment:
Should this reference the `service.httpPort`?

Reply:
A lot of time in charts the they fix the ports of the containers, and then only make the service ports configurable but I never really liked that convention. I went w/ your suggestions so that the service ports match what the container listens on, and the values are reused to template the service, pod, and application configs. Thanks!